ci/infra/testrunner/skuba/skuba.py

```diff
@@ -91,9 +91,15 @@ def node_join(self, role="worker", nr=0):
 
     def join_nodes(self, masters=None, workers=None):
         if masters is None:
-            masters = self.platform.get_num_nodes("master")
+            if self.conf.skuba.masters is not None:
```

Comment:
I think the parameters should take precedence over the config, and log when there is an override from platform.get_num_nodes (since in theory that call returns all available nodes to join)

Reply:
I don't see the purpose of the `skuba.master` variable. You should use either the number of provisioned nodes or the number requested in this function.